rdtools/filtering.py

```diff
@@ -124,3 +140,322 @@ def csi_filter(poa_global_measured, poa_global_clearsky, threshold=0.15):
+    ----------
+    power_ac : pd.Series
+        Pandas time series, representing PV system power or energy with
+        a pandas datetime index.
+    roll_periods: Int
+        Number of readings to calculate the rolling maximum derivative on.
+    Returns
+    -------
+    pd.Series
+        Time series of the rolling maximum derivative.
+    """
+    # Calculate the maximum value over a forward-rolling window
+    max_roll = power_ac.iloc[::-1].rolling(roll_periods).max()
+    max_roll = max_roll.reindex(power_ac.index)
+    # Calculate the minimum value over a forward-rolling window
+    min_roll = power_ac.iloc[::-1].rolling(roll_periods).min()
+    min_roll = min_roll.reindex(power_ac.index)
+    # Calculate the maximum derivative within the foward-rolling window
+    derivative_max = (max_roll - min_roll) / ((max_roll + min_roll) / 2) * 100
+    return derivative_max
+
+
+def logic_clip_filter(power_ac,
+                      mounting_type='Fixed',
+                      derivative_cutoff=0.2,
+                      roll_periods=None):
+    '''
+    This filter is a logic-based filter that is used to filter out
+    clipping periods in AC power and AC energy time series.
+    The AC power or energy time series is filtered based on the
+    maximum derivative over a rolling window, as compared to a user-set
+    derivate_cutoff (default set to 0.2).  The size of the
+    rolling window is increased when the system is a tracked system.
+    Parameters
+    ----------
+    power_ac : pd.Series
+        Pandas time series, representing PV system power or energy with
+        a pandas datetime index.
+    mounting_type : String
+        String representing the mounting configuration associated with the
+        AC power/energy time series. Can either be "Fixed" or "Tracking".
+        Default set to 'Fixed'.
+    derivative_cutoff : Float
+        Cutoff for max derivative threshold. Defaults to 0.2; however, values
+        as high as 0.4 have been tested and shown to be effective.  The higher
+        the cutoff, the more values in the dataset that will be determined as
+        clipping.
+    roll_periods: Integer.
+        Number of periods to examine when looking for a near-zero derivative
+        in the time series derivative. If roll_periods = 3, the system looks
+        for a near-zero derivative over 3 consecutive readings. Default value
+        is set to None, so the function uses default logic: it looks for a
+        near-zero derivative over 3 periods for a fixed tilt system, and over
+        5 periods for a tracked system with a sampling frequency more frequent
+        than once every 30 minutes.
+    Returns
+    -------
+    pd.Series
+        The passed time series, filtered to exclude clipping periods.
+    pd.Series
+        Boolean Series of whether to include the point because it is not
+        clipping.
+        True values delineate non-clipping periods, and False values delineate
+        clipping periods.
```

Comment:
Seems like the boolean series is really the primary output here, as it is used to filter more than just the input power series.  Should we remove the filtered time series return value?  It's easy for the user to calculate it themselves, and it would make this function's return value consistent with the existing filtering functions.